File: samples/Android/xtruntime/src/main/java/com/opensource/xtruntime/XTRDevice.kt
package com.opensource.xtruntime

import android.os.Build

/**
 * Created by cuiminghui on 2017/9/28.
 */
class XTRDevice: XTRComponent() {

    override val name: String = "XTRDevice"

    private var current: InnerObject? = null

    fun xtr_current(): InnerObject {
        if (current == null) {
            current = InnerObject(xtrContext)
        }
        return current ?: InnerObject(xtrContext)
    }

    class InnerObject(val xtrContext: XTRContext) {

        fun xtr_name(): String {
            return Build.BRAND
        }

        fun xtr_systemName(): String {
            return "Android"
        }

        fun xtr_systemVersion(): String {
            return Build.VERSION.SDK_INT.toString()
        }

        fun xtr_xtRuntimeVersion(): String {
            return XTRuntime.version
        }

        fun xtr_model(): String {
            return Build.MODEL
        }

        fun xtr_orientation(): Int {
            return 0
        }

    }

}

File: src/implementation/ios/runtime/XTRDevice.m
//
//  XTRDevice.m
//  XTSample
//
//  Created by 崔明辉 on 2017/9/28.
//  Copyright © 2017年 UED Center, YY Inc. All rights reserved.
//

#import "XTRDevice.h"
#import "XTRuntime.h"

@implementation XTRDevice

+ (NSString *)name {
    return @"XTRDevice";
}

+ (XTRDevice *)xtr_current {
    return [XTRDevice new];
}

- (instancetype)init
{
    self = [super init];
    if (self) {
        self.objectUUID = [[NSUUID UUID] UUIDString];
    }
    return self;
}

- (NSString *)xtr_name {
    return [UIDevice currentDevice].name;
}

- (NSString *)xtr_systemName {
    return [UIDevice currentDevice].systemName;
}

- (NSString *)xtr_systemVersion {
    return [UIDevice currentDevice].systemVersion;
}

- (NSString *)xtr_xtRuntimeVersion {
    return [XTRuntime version];
}

- (NSString *)xtr_model {
    return [UIDevice currentDevice].model;
}

- (NSInteger)xtr_orientation {
    return [UIDevice currentDevice].orientation;
}

@end